### PR TITLE
Disabled http protocol for remote URL

### PR DESCRIPTION
**Pull request by jalalkhademi**

### CODE REVIEW

**Comment by jalalkhademi** [comment]
I added that attribute to xml file and parse it in UrlInputPreference class.

**Comment by maniac103** [comment]
Please use an attribute resource for that. `ItemUpdatingPreference` can serve as an example here.
Additionally, the existing error message should be either kept unchanged or the protocol be inserted via positional argument, as http is still valid in local case.

**Comment by jalalkhademi** [comment]
@maniac103 Many thanks for your fast and helpful review. I fixed the issues.

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 1]
```diff
@@ -116,8 +130,8 @@
         companion object {
             private const val KEY_TITLE = "title"
 
-            fun newInstance(key: String, title: CharSequence): PrefFragment {
-                val f = PrefFragment()
+            fun newInstance(key: String, title: CharSequence, isHttpEnabled: Boolean): PrefFragment {
+                val f = PrefFragment(isHttpEnabled)
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: Fragments must not have constructor parameters. Just put the boolean into the arguments bundle as well.

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 2]
```diff
@@ -89,15 +100,18 @@
                         val url = URL(value)
                         urlIsValid = true
                         when (url.protocol) {
-                            "http" -> portSeemsInvalid = url.port == 443 || url.port == 8443
                             "https" -> portSeemsInvalid = url.port == 80 || url.port == 8080
+                            "http" -> when
+                                    {
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: Curly brace should be on old line. Also maybe could use an if here?

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 3]
```diff
@@ -78,6 +88,7 @@
 
         override fun afterTextChanged(editable: Editable) {
             var portSeemsInvalid = false
+            val res = editor.resources
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: This variable movement seems unneeded?

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 4]
```diff
@@ -89,15 +100,18 @@
...
+                                    }
                         }
                     } catch (e: MalformedURLException) {
                         urlIsValid = false
                     }
                 }
             }
-            val res = editor.resources
             wrapper.error = when {
                 !urlIsValid -> res.getString(R.string.error_invalid_url)
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: Need to distinguish strings between 'http allowed' and 'http not allowed' here.

**Comment by mueller-ma** [comment]
Currently two required checks are failing. You need to configure your git cli name and mail to match those in GitHub.
`git config --local user.name jalalkhademi && git config --local user.email jalal.khademi66@gmail.com`
After that you must change the author of the commits. My preferred way is:
`git reset --soft HEAD~4` (`4` because of 4 commits in this PR). `git add . && git commit -s -m "Disabled http protocol for remote URL"`. `git push --force-with-lease`.

**Comment by jalalkhademi** [comment]
@maniac103 fixed all mentioned points. Just I have no idea what is the process for updating translated strings.

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 5]
```diff
@@ -33,8 +33,17 @@ import java.net.URL
 class UrlInputPreference constructor(context: Context, attrs: AttributeSet) :
     CustomInputTypePreference(context, attrs) {
 
+    var isHttpEnabled = false
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: Oops, sorry, I missed this earlier: this should be a val, I think.

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 6]
```diff
@@ -45,6 +54,7 @@ class UrlInputPreference constructor(context: Context, attrs: AttributeSet) :
         private lateinit var wrapper: TextInputLayout
         private lateinit var editor: EditText
         private var urlIsValid: Boolean = false
+        private var isHttpEnabled = false
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: No need for a member for this, just use a local variable.

**Comment by jalalkhademi** [comment]
Those two points are also resolved now

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt [hunk 7]
```diff
@@ -33,8 +33,11 @@ import java.net.URL
 class UrlInputPreference constructor(context: Context, attrs: AttributeSet) :
     CustomInputTypePreference(context, attrs) {
 
+    private val isHttpEnabled = context.obtainStyledAttributes(attrs, R.styleable.UrlInputPreference)
+        .getBoolean(R.styleable.UrlInputPreference_isHttpEnabled, false)
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/UrlInputPreference.kt]: This must stay in the init block for the recycle() call.

**Comment by jalalkhademi** [comment]
Done

**Comment by jalalkhademi** [comment]
@mueller-ma Done.

**Comment by mueller-ma** [comment]
Thanks. Will be merged tomorrow (https://github.com/openhab/openhab-android/issues/549#issuecomment-547916527).

**Pull request merged**